__1. У коробці знаходяться зелені, жовті та сині м’ячі. Вірогідність витягнути навмання 
не зелений м’яч дорівнює 0.6, а вірогідність витягнути навмання не жовтий м’яч 
дорівнює 0.55. Яка вірогідність витягнути навмання не синій м’яч?
У скільки разів жовтих м’ячів більше, ніж синіх.__

In [138]:
# вероятности не достать мячики
not_green = 0.6 
not_yellow = 0.55

# из уравнения | P(not_green) + P(not_yellow) + P(not_blue) = 2 | найдем вероятность не достать синий мячик
not_blue = 2 - not_green - not_yellow
not_blue

0.8499999999999999

In [139]:
# вероятность достать мячик:
green = 1 - not_green # 0.4
yellow = 1 - not_yellow # 0.45
blue = 1 - not_blue # 0.15

# соотношение количества желтых мячиков к количеству синих мячиков
yellow_to_blue = yellow / blue
yellow_to_blue

2.999999999999997

Ответ: 

- Вероятность не вытянуть синий мячик = 0.15

- Желтых мячиков больше чем синих в 3 раза

----------------------

__2.Людина стріляє по п’яти цілям. Вірогідність влучення по будь-якій цілі дорівнює 3/5
Яка вірогідність того, що з 5 цілей рівно 3 будуть уражені?__

In [140]:
import math
n = 5
p = 3/5 # вероятность попадания
k = 3
# используем формулу биномиального распределения, где | math.comb(n, k) | - это биномиальный коэффициент
P = math.comb(n, k)*(p**k)*((1-p)**(n-k))
P

0.3456

Ответ:

- Вероятность того что из 5 целей в 3 попадут = 0.34

----------------------

__3.Проводиться експеримент з підкиданням монетки. Ціль експерименту – отримати 5 
підкидань решки підряд. 
Якщо при підкиданні монетки випадає орел – експеримент завершується, і 
починається новий, до тих пір, доки не буде досягнута ціль експерименту. 
Наведіть вірогідності всіх можливих результатів даного експерименту. 
Яка кількість решок випадає в середньому під час одного експерименту?__

In [141]:
# Вероятность выпадения решки = 0.5, вероятность выпадение решки 5 раз подряд = 0.5 в пятой степени
(0.5)**5

0.03125

In [142]:
import numpy as np
# симуляция броска монеты пока не выпадет 5 решек подряд
def coin_flip():
    count_reshka = 0 # к-во выпавши решек 
    total_flip = 0 # к-во бросков
    
    while count_reshka < 5:
        total_flip += 1 
        flip = np.random.binomial(1, 0.5) # 1 = решка, 0 = орел
        if flip == 1:
            count_reshka += 1
        else:
            count_reshka = 0
            
    return total_flip

In [143]:
#повторим эксперимент 100к раз, что бы найти среднее к-во выпавших решек

total_reshka = 0 # к-во решек за все эксперименты
k = 100000 # к-во повторений эксперимента

for _ in range(k):
    total_reshka += coin_flip()
    
total_reshka / k


61.75398

Ответ:

- Вероятность выпадения решки 5 раз подряд = 0.03

- Среднее количество выпадения решки за 1 эксперименты ≈ 62

----------------------

__4. Є механічний «однорукий бандит» з трьома барабанами.
Стрічки барабанів і таблиця виплат наведені нижче.
Гравець має можливість зробити ставку 1 грн і отримати виграш, якщо випала «в 
лінію» одна з трьох комбінацій, наведених в таблиці виплат.
Всі комбінації рахуються зліва направо, починаючи з крайнього лівого барабана 
(комбінація 22 означає, що на перших двох барабанах має випасти символ 2). 
Виплата здійснюється лише за старшу комбінацію (якщо на всіх трьох барабанах 
випадає символ 2, то отримаємо виграш лише за комбінацію 222).
Який відсоток грошей буде виплачуватися гравцям на нескінченно довгому 
проміжку часу?
Як часто буде випадати будь-який виграш?__

Барабан 1:  | 1 | 2 | 3 | 4 | 5 | 1 | 2 | 3 | 4 | 5 |

Барабан 2:  | 1 | 2 | 3 | 4 | 5 | 1 | 2 | 3 | 4 | 5 |

Барабан 3:  | 3 | 1 | 5 | 2 | 3 | 2 | 4 | 1 | 3 | 2 |

In [145]:
# к-во возможных комбиниаций
N = 10**3 # 1000

# вероятности выпадений комбинаций   !!На 3м барабане "2" и "3" встречаются 3 раза!!
P111 = 2/10 * 2/10 * 2/10 # 0.004
P11 = 2/10 * 2/10 #0.04

P222 = 2/10 * 2/10 * 3/10 #0.012
P22 = 2/10 * 2/10 #0.04

P333 = 2/10 * 2/10 * 3/10 #0.012

# вероятность выигрыша = сумма вероятностей всех комбинаций
P_vin = P111 + P11 + P222 + P22 + P333

# ожидаемая выплата = сумме произведений номинала выигрыша на вероятность комбинации
P_pay = (50 * P111) + (5 * P11) + (20 * P222) + (3 * P22) + (5 * P333)

print('P_vin:', P_vin, '\nP_pay:', P_pay, 'грн')


P_vin: 0.112 
P_pay: 1.02 грн


Ответ:
- 102% денег, будет выплачиватся на бесконечную игру

- Вероятность выигрыша = 11.2%